torch/_inductor/lowering.py

```diff
@@ -292,11 +292,18 @@ def broadcast_symbolic_shapes(a, b):
 def promote_constants(inputs, override_return_dtype=None):
     if not any(isinstance(x, (sympy.Expr, int, float)) for x in inputs):
         return inputs
-    if all(isinstance(x, (int, float)) for x in inputs):
+    if all(isinstance(x, (int, float, sympy.Symbol)) for x in inputs):
         dtype = override_return_dtype or get_promoted_dtype(
             *inputs, type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
         )
-        return [ir.Constant(x, dtype, decode_device(None)) for x in inputs]
+
+        def const_func(x):
+            if isinstance(x, sympy.Symbol):
+                return ir.IndexingConstant(x, dtype, decode_device(None))
```

Comment:
Same `ir` class that's used for `sympy.Expr` below, with params from `Constant`.

Reply:
Other changes in the PR are due to `promote_constants`, which is called from `make_pointwise` (a lowering decorator).